src/Dijkstra/AST/Examples/PaperIntro.agda

```diff
@@ -0,0 +1,165 @@
+                         (Dijkstra.AST.Core.AST.ASTop (Left (RWSBase.RWStell (a ∷ []) refl))
+                          (λ ()))
+                         (λ _ → Dijkstra.AST.Core.AST.ASTreturn (unit , (λ _ → [])))
+                   })))
+              (proj₁ i , proj₂ i)))
+            (proj₂
+             (proj₂
+              (Dijkstra.AST.Core.ASTOpSem.runAST RWSBase.RWSOpSem
+               (Dijkstra.AST.Branching.ASTExtension.unextend RWSBase.RWSOps
+                (Dijkstra.AST.Core.AST.ASTop
+                 (Right (Dijkstra.AST.Branching.BranchCmd.BCmaybe (f (proj₂ i))))
+                 (λ { (Level.lift nothing)
+                        → Dijkstra.AST.Core.AST.ASTreturn (unit , (λ x₁ → x₁ ++ x₁))
+                    ; (Level.lift (just a))
+                        → Dijkstra.AST.Core.AST.ASTbind
+                          (Dijkstra.AST.Core.AST.ASTop (Left (RWSBase.RWStell (a ∷ []) refl))
+                           (λ ()))
+                          (λ _ → Dijkstra.AST.Core.AST.ASTreturn (unit , (λ _ → [])))
+                    })))
+               (proj₁ i , proj₂ i))))))
+
+  -- This is quite unwieldy, and it's very challenging to look at this and determine that
+  -- we need to do 'with f s' (after refining the input i to (_ , s)), to get the two cases to
+  -- prove.  And this is for a small, relatively simple program that is all in scope, so we have
+  -- no need to prove and invoke properties about functions called within the program.
+
+  -- The proof via predicate transformer semantics and sufficiency that follow demonstrates how
+  -- much easier the framework makes it to prove this property.
+
+-}
+
+-- See below for step-by-step development and explanation of this proof
+progPostWP : ∀ f i → predTrans (prog f) (ProgPost i) i
+proj₁ (progPostWP _ _ _ refl) _ = λ where          _ refl → refl , refl
+proj₂ (progPostWP _ _ _ refl) j = λ where _ _ refl _ refl → refl , refl
+
+{-
+  -- We start with:
+
+  progPostWP : ∀ f i → predTrans (prog f) (ProgPost i) i
+  progPostWP f i = {!!}
+
+  -- In the following, each time we make a step, the goal in each hole is much more manageable and
+  -- understandable than the unwieldy one shown above for the direct approach.
+
+  -- C-c C-c in *empty* hole, press enter to split on result
+
+  progPostWP f i r x = {!!}
+
+  -- C-c C-c with x in the hole
+
+  progPostWP f i .(f (proj₂ i)) refl = {!!}
+
+  -- C-c C-c in *empty* hole, press enter to split on result
+
+  proj₁ (progPostWP f i .(f (proj₂ i)) refl) = {!!}
+  proj₂ (progPostWP f i .(f (proj₂ i)) refl) = {!!}
+
+  -- C-c C-r in *each* hole
+
+  proj₁ (progPostWP f i .(f (proj₂ i)) refl) = λ        x o' x₁ → {!!}
+  proj₂ (progPostWP f i .(f (proj₂ i)) refl) = λ j x r x₁ o' x₂ → {!!}
+
+  -- x₁ : o' ≡  ([] ++ []) ++ [] ++ [], change to λ where ... refl to rewrite it,
+  -- and similarly for x₂ : o' ≡ []
+
+  proj₁ (progPostWP f i .(f (proj₂ i)) refl) = λ where       x  o' refl → {!!}
```

Comment:
Thanks.  I think :smile:.  The alignment made me understand something that I may not have understood before, and resulted in me adding a lot more detail to the example.  See #239.

Reply:
I love alignment - it really helps see how each part fits into the bigger picture.
And I love naming variables names that really say something.
It makes proofs more readable (rather than write-only).